### PR TITLE
feat: add prefetch support to `InvertedIndexFooterReader` for reduced I/O time

### PR DESCRIPTION
**Pull request by WenyXu**
I hereby agree to the terms of the [GreptimeDB CLA](https://github.com/GreptimeTeam/.github/blob/main/CLA.md).

## What's changed and what's your intention?

Add prefetch support to `InvertedIndexFooterReader`, and set the default prefetch size to 1KiB. To reduce the inefficiency of discrete small S3 IO operations.

1280X1280

## Checklist

- [ ] I have written the necessary rustdoc comments.
- [ ] I have added the necessary unit tests and integration tests.
- [ ] This PR requires documentation updates.

### CODE REVIEW

File: src/index/src/inverted_index/format/reader/footer.rs [hunk 1]
```diff
@@ -18,53 +18,83 @@ use prost::Message;
...
 
-    async fn read_payload_size(&mut self) -> Result<u64> {
-        let mut size_buf = [0u8; FOOTER_PAYLOAD_SIZE_SIZE as usize];
-        let end = self.blob_size;
-        let start = end - FOOTER_PAYLOAD_SIZE_SIZE;
-        self.source
-            .read_into(start..end, &mut &mut size_buf[..])
+impl<R: RangeReader> InvertedIndexFooterReader<R> {
+    pub async fn metadata(&mut self) -> Result<InvertedIndexMetas> {
+        let footer_start = self.blob_size.saturating_sub(self.prefetch_size());
```
Comment (v0y4g3r) [inline src/index/src/inverted_index/format/reader/footer.rs:58]: nit: maybe we can direrct raise an error here in that blob size is less than FOOTER_PAYLOAD_SIZE_SIZE, so we can spare a extra IO when we know the file is corrupted.

**Review by v0y4g3r — APPROVED** [review]
Rest LGTM

**Review by zhongzc — APPROVED** [review]
LGTM

File: src/index/src/inverted_index/format/reader/footer.rs (hunk 1 above)
Comment (WenyXu) [reply src/index/src/inverted_index/format/reader/footer.rs:63]: resolved in [c14ab76](https://github.com/GreptimeTeam/greptimedb/pull/5146/commits/c14ab76e546e16326f4a1fdc6b48ea017a810eb4)

**Pull request merged**